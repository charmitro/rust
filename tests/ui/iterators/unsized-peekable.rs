//@ compile-flags: -Zmir-enable-passes=+Inline -Zmir-enable-passes=+JumpThreading --crate-type=lib

pub fn problem_thingy(items: &mut impl Iterator<Item = str>) {
    let mut peeker = items.peekable();
    //~^ ERROR: the size for values of type `str` cannot be known at compilation time [E0277]
    // ^^^ doesn't have a size known at compile-time
    match peeker.peek() {
        Some(_) => (),
        None => return (),
    }
}
