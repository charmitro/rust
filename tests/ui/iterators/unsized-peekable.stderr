error[E0277]: the size for values of type `str` cannot be known at compilation time
  --> $DIR/unsized-peekable.rs:4:28
   |
LL |     let mut peeker = items.peekable();
   |                            ^^^^^^^^ doesn't have a size known at compile-time
   |
   = help: the trait `Sized` is not implemented for `str`
note: required by a bound in `peekable`
  --> $SRC_DIR/core/src/iter/traits/iterator.rs:LL:COL

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0277`.
